### ISSUE TITLE
`DELETE` endpoints with `requestBody` leads to an OpenAPI semantic error

### ISSUE THREAD

**Issue by McMerph** [RobinTail/express-zod-api#820]
It looks like OpenAPI spec forbids a request body for `DELETE` operations:
- According to https://www.rfc-editor.org/rfc/rfc9110.html#name-delete:
> Although request message framing is independent of the method used, content received in a DELETE request has no generally defined semantics, cannot alter the meaning or target of the request, and might lead some implementations to reject the request and close the connection because of its potential as a request smuggling attack
- According to https://stackoverflow.com/a/54980985:
> No, you cannot use the OpenAPI 3.0 Specification and Swagger tools to implement DELETE requests with a request body

---
### Steps to reproduce:
- add endpoint with `delete` method like this:
```
defaultEndpointsFactory.build({
  methods: ["delete"],
  input: z.object({}),
...
```
- call `new OpenAPI({...}).getSpecAsYaml()`
- validate generated spec (e.g. on https://editor.swagger.io/)

### Actual behaviour:
There is a semantic error `DELETE operations cannot have a requestBody` in OpenAPI spec

### Expected behaviour
OpenApi spec generates without `requestBody` for DELETE operations

**Comment by RobinTail** [comment]
Okay. I agree.

**Comment by RobinTail** [comment]
I'm going to include your fix in the next major release, @McMerph .
Along with another potentially breaking feature.
There will be beta as well, which you could use for your needs.

**Comment by RobinTail** [comment]
The fix released in version `v9.0.0-beta1`, @McMerph
